### PR TITLE
Refactoring/file storage

### PR DESCRIPTION
**Pull request by mathieutrudeau**
This pull request introduces significant improvements to our file management system, primarily driven by the addition of a File Storage Helper. The purpose of this helper is to streamline file operations within our application and enhance the system's robustness. Additionally, it incorporates a crucial Docker Compose configuration update to ensure data integrity.

### CODE REVIEW

**Review by mubashir494 — APPROVED** [review]
Lgtm

File: Backend/Backend/Infrastructure/FileStorageHelper.cs [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+        // Return the filename stored in the database
+        return userAvatarName;
+    }
+
+    /// <summary>
+    /// Removes a user avatar.
+    /// </summary>
+    /// <param name="userAvatarName"></param>
+    public static void RemoveUserAvatar(string userAvatarName)
```
Comment (BigSaladMan) [inline Backend/Backend/Infrastructure/FileStorageHelper.cs:82]: Could this potentially delete a placeholder one? like the default placeholder one

File: Backend/Backend/Infrastructure/FileStorageHelper.cs [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+        // Filename stored on the server filesystem
+        string coverArtFileName = string.Format("{0}.{1}", podcastId, coverArtFile.ContentType.Split('/')[1]);
+        
+        // Filename stored in the database
+        string coverArtName = string.Format("{0}{1}{2}", coverArtFileName, FILE_SPLIT_KEY, coverArtFile.ContentType);
+
+        // Get the dir path
+        string dirPath = Combine(GetCurrentDirectory(), BASE_DIR, PODCASTS_DIR_NAME);
+
+        // Make sure that the dir exists, otherwise create it
```
Comment (BigSaladMan) [inline Backend/Backend/Infrastructure/FileStorageHelper.cs:137]: is this necessary? we have another one right below

File: Backend/Backend/Infrastructure/FileStorageHelper.cs [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+    public const string AVATARS_DIR_NAME = "Avatars";
+
+    /// <summary>
+    /// Dir where all podcast covers and episodes are stored. 
+    /// </summary>
+    public const string PODCASTS_DIR_NAME = "Podcasts";
+
+    #region User Profile
+
+    /// <summary>
```
Comment (BigSaladMan) [inline Backend/Backend/Infrastructure/FileStorageHelper.cs:35]: I understand why you did this, but its kinda redundant. why not just use a toString() in the original call?

**Review by BigSaladMan — APPROVED** [review]

File: Backend/Backend/Infrastructure/FileStorageHelper.cs (hunk 2 above)
Comment (BigSaladMan) [reply Backend/Backend/Infrastructure/FileStorageHelper.cs:137]: oh that didnt comment correctly, I meant the create dir right below this

File: Backend/Backend/Infrastructure/FileStorageHelper.cs (hunk 3 above)
Comment (BigSaladMan) [reply Backend/Backend/Infrastructure/FileStorageHelper.cs:35]: ...same with this one. I mean to comment on the method
public static string SaveUserAvatar(Guid userId, IFormFile avatarFile)
    {
        return SaveUserAvatar(userId.ToString(), avatarFile);
    }

File: Backend/Backend/Infrastructure/FileStorageHelper.cs (hunk 1 above)
Comment (mathieutrudeau) [reply Backend/Backend/Infrastructure/FileStorageHelper.cs:82]: No, the placeholder would not be saved on the server.

File: Backend/Backend/Infrastructure/FileStorageHelper.cs (hunk 3 above)
Comment (mathieutrudeau) [reply Backend/Backend/Infrastructure/FileStorageHelper.cs:35]: For ease of use.

File: Backend/Backend/Infrastructure/FileStorageHelper.cs (hunk 2 above)
Comment (mathieutrudeau) [reply Backend/Backend/Infrastructure/FileStorageHelper.cs]: You're right. Removed it.

**Pull request merged**